### PR TITLE
2496 bad arg exception message

### PR DESCRIPTION
**Pull request by TyOverby**
addresses https://github.com/dotnet/roslyn/issues/2496

### CODE REVIEW

File: src/Compilers/Core/Portable/Text/SourceTextStream.cs [hunk 1]
```diff
@@ -80,7 +80,7 @@ public override int Read(byte[] buffer, int offset, int count)
                 // The buffer must be able to hold at least one character from the 
                 // SourceText stream.  Returning 0 for that case isn't correct because
                 // that indicates end of stream vs. insufficient buffer. 
-                throw new ArgumentException(nameof(count));
+                throw new ArgumentException($"{nameof(count)} must be less than {_minimumTargetBufferCount}", nameof(count));
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Text/SourceTextStream.cs]: strictly less than

Comment (TyOverby) [inline src/Compilers/Core/Portable/Text/SourceTextStream.cs]: Done

**Comment by TyOverby** [comment]
Thanks @jcouv!

@dotnet/roslyn-compiler 

One more review please!

File: src/Compilers/Core/Portable/Text/SourceTextStream.cs [hunk 2]
```diff
@@ -80,7 +80,7 @@ public override int Read(byte[] buffer, int offset, int count)
                 // The buffer must be able to hold at least one character from the 
                 // SourceText stream.  Returning 0 for that case isn't correct because
                 // that indicates end of stream vs. insufficient buffer. 
-                throw new ArgumentException(nameof(count));
+                throw new ArgumentException($"{nameof(count)} must be strictly less than {_minimumTargetBufferCount}", nameof(count));
```
Comment (cston) [inline src/Compilers/Core/Portable/Text/SourceTextStream.cs]: `cannot be less than...`?

Comment (jcouv) [inline src/Compilers/Core/Portable/Text/SourceTextStream.cs]: Ah, you're right. I read the condition in reverse :-S

Comment (TyOverby) [inline src/Compilers/Core/Portable/Text/SourceTextStream.cs]: Done

**Comment by cston** [comment]
LGTM

**Comment by AlekseyTs** [comment]
@TyOverby It feels like the messages should not be hardcoded, but instead should be put in resource file so that they could be localized.

**Pull request merged**